src/tiffcomposite_int.cpp

```diff
@@ -473,8 +473,11 @@ TiffComponent* TiffDirectory::doAddPath(uint16_t tag, TiffPath& tiffPath, TiffCo
     return tc->addPath(tag, tiffPath, pRoot, std::move(object));
 
   auto atc = [&] {
-    if (tiffPath.size() == 1 && object)
-      return std::move(object);
+    if (tiffPath.size() == 1 && object) {
+      TiffComponent::UniquePtr tempObject;
+      std::swap(object, tempObject);
```

Comment:
auto tempObject = std::move(object);
return tempObject

Reply:
I actually wonder about std::move() in the original code. It should warn with -Wpessimising-move, but it doesn't.

Reply:
Interesting. CTAD doesn't apply to lambdas.

Reply:
my suggestion still triggers the warning.

Reply:
Yes, I think clang-tidy is just detecting that there's a control-flow path with two consecutive `std::move` calls. I tricked it by replacing the `std::move` with a `std::swap`. My solution doesn't improve the code, it just makes the warning go away.